From d2dd5ae63d2d5e598dd76d075ea998b4d3062f13 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Sat, 13 Feb 2010 15:25:39 +0000
Subject: [PATCH] setup-disk: boot fix for mdadm-3.1.1

mdadm 3.1.1 switched to use metadata 1.1 by default. This does not work
with extlinux so we set metadata to old default manually for /boot.
---
 setup-disk.in |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index a6b9efa..a81172a 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -327,13 +327,15 @@ EOF
 		local p= rd=
 		for p in $(sfdisk -l $rootdisk_dev 2>/dev/null \
 				| awk '/Linux raid/ {print $1}'); do
+			local opt="--metadata=0.90"
 			case "$p" in
 				*1) rd=/dev/md0; boot_dev=/dev/md0;;
-				*2) rd=/dev/md1; swap_dev=/dev/md1;;
+				*2) rd=/dev/md1; swap_dev=/dev/md1
+				    opt= ;;
 				*3) rd=/dev/md2; root_dev=/dev/md2;;
 			esac
 			mdadm --create $rd --level=1 --raid-devices=2 \
-				--quiet --run $p missing
+				$opt --quiet --run $p missing
 		done
 	else
 		local p=
-- 
1.6.6.1

